suite-native/accounts/src/components/AccountListItem.tsx

```diff
@@ -1,75 +1,92 @@
 import { Account } from '@suite-common/wallet-types';
-import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { Badge, Box, HStack, RoundedIcon, Text } from '@suite-native/atoms';
 import { CryptoAmountFormatter, CryptoToFiatAmountFormatter } from '@suite-native/formatters';
-import { selectIsEthereumAccountWithTokensWithFiatRates } from '@suite-native/ethereum-tokens';
+import { Translation } from '@suite-native/intl';
 import { SettingsSliceRootState } from '@suite-native/settings';
+import { isCoinWithTokens, selectNumberOfAccountTokensWithFiatRates } from '@suite-native/tokens';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
 
 export type AccountListItemProps = {
     account: Account;
-    areTokensDisplayed?: boolean;
+    hideTokens?: boolean;
 };
 
-const accountListItemStyle = prepareNativeStyle<{ isFollowedByTokens: boolean }>(
-    (_, { isFollowedByTokens }) => ({
-        flexDirection: 'row',
-        justifyContent: 'space-between',
-        alignItem: 'center',
-        extend: {
-            condition: isFollowedByTokens,
-            style: {
-                paddingBottom: 0,
-            },
-        },
-    }),
-);
+const accountListItemStyle = prepareNativeStyle(utils => ({
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItem: 'center',
+
+    paddingVertical: 12,
+    paddingHorizontal: utils.spacings.medium,
+}));
 
-export const accountDescriptionStyle = prepareNativeStyle(_ => ({
+const accountDescriptionStyle = prepareNativeStyle(_ => ({
     flexShrink: 1,
 }));
 
-export const valuesContainerStyle = prepareNativeStyle(utils => ({
+const valuesContainerStyle = prepareNativeStyle(utils => ({
     maxWidth: '40%',
     flexShrink: 0,
     alignItems: 'flex-end',
     paddingLeft: utils.spacings.small,
 }));
 
-export const AccountListItem = ({ account, areTokensDisplayed = false }: AccountListItemProps) => {
+export const AccountListItem = ({ account, hideTokens = false }: AccountListItemProps) => {
     const { applyStyle } = useNativeStyles();
     const { accountLabel } = account;
+    const { NetworkNameFormatter } = useFormatters();
 
     const formattedAccountType = useSelector((state: AccountsRootState) =>
         selectFormattedAccountType(state, account.key),
     );
-
-    const isAccountWithTokens = useSelector(
-        (state: FiatRatesRootState & SettingsSliceRootState & AccountsRootState) =>
-            selectIsEthereumAccountWithTokensWithFiatRates(state, account.key),
+    const numberOfTokens = useSelector(
+        (state: AccountsRootState & FiatRatesRootState & SettingsSliceRootState) =>
+            selectNumberOfAccountTokensWithFiatRates(state, account.key),
     );
 
+    const doesCoinSupportTokens = isCoinWithTokens(account.symbol);
+    const shouldShowAccountLabel = !doesCoinSupportTokens || hideTokens;
+    const shouldShowTokenBadge = numberOfTokens > 0 && hideTokens;
```

Comment:
Negation possibly missing here?
```suggestion
    const shouldShowTokenBadge = numberOfTokens > 0 && !hideTokens;
```

Reply:
It's correct it should show badge only when tokens are hidden in modal. I am thinking about renaming it to `hideTokenToModal` or something like that.